### ISSUE TITLE
VSCode Extension: Client onyx-lsp: connection to server is erroring. Shutting down server.

### ISSUE THREAD

**Issue by magnetenstad** [onyx-lang/onyx#33]
Steps to reproduce:
1. Download [onyx-windows-wasmer-amd64.zip](https://github.com/onyx-lang/onyx/releases/download/v0.1.8-beta/onyx-windows-wasmer-amd64.zip)
2. Place the dist folder in `C:` and rename it to `onyx`
3. Add `C:/onyx` to Path and add `ONYX_PATH`: `C:/onyx`
4. Install [the extension](https://marketplace.visualstudio.com/items?itemName=onyxlang.onyxlang)
5. Open a `.onyx` file in VSCode. Extension now complains that there is no path `C:/onyx/bin`
6. Move `onyx.exe` from `C:/onyx` to `C:/onyx/bin` and edit Path from `C:/onyx` to `C:/onyx/bin`
7. Open a `.onyx` file in VSCode. Extension now gives the error below

Output:
```
[Error - 12:40:52 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Info  - 12:40:52 AM] Connection to server got closed. Server will restart.
true
[Error - 12:40:52 AM] onyx-lsp client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 12:40:52 AM] Client onyx-lsp: connection to server is erroring. Shutting down server.
[Error - 12:40:52 AM] Stopping server failed
Error: Client is not running and can't be stopped. It's current state is: startFailed
	at LanguageClient.shutdown (c:\Users\tenst\.vscode\extensions\onyxlang.onyxlang-0.1.8\node_modules\vscode-languageclient\lib\common\client.js:914:19)
	at LanguageClient.stop (c:\Users\tenst\.vscode\extensions\onyxlang.onyxlang-0.1.8\node_modules\vscode-languageclient\lib\common\client.js:885:21)
	at LanguageClient.stop (c:\Users\tenst\.vscode\extensions\onyxlang.onyxlang-0.1.8\node_modules\vscode-languageclient\lib\node\main.js:150:22)
	at LanguageClient.handleConnectionError (c:\Users\tenst\.vscode\extensions\onyxlang.onyxlang-0.1.8\node_modules\vscode-languageclient\lib\common\client.js:1146:18)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Error - 12:40:52 AM] Server initialization failed.
  Message: write EPIPE
  Code: -32099 
[Error - 12:40:52 AM] onyx-lsp client: couldn't create connection to server.
  Message: write EPIPE
  Code: -32099 
[Error - 12:40:52 AM] Restarting server failed
  Message: write EPIPE
  Code: -32099 
[Info  - 12:40:52 AM] Connection to server got closed. Server will restart.
true
[Error - 12:40:52 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Info  - 12:40:52 AM] Connection to server got closed. Server will restart.
true
[Error - 12:40:52 AM] onyx-lsp client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 12:40:52 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Info  - 12:40:52 AM] Connection to server got closed. Server will restart.
true
[Error - 12:40:52 AM] onyx-lsp client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 12:40:52 AM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 12:40:53 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Info  - 12:40:53 AM] Connection to server got closed. Server will restart.
true
[Error - 12:40:53 AM] onyx-lsp client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 12:40:53 AM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 12:40:53 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 12:40:53 AM] The onyx-lsp server crashed 6 times in the last 3 minutes. The server will not be restarted. See the output for more information.
[Error - 12:40:53 AM] onyx-lsp client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 12:40:53 AM] Restarting server failed
  Message: Pending response rejected since connection got disposed
  Code: -32097 
```

**Comment by brendanfh** [comment]
I will preface this saying that I don't run Windows daily and I haven't tested the LSP on Windows in a long time. I know the LSP is looking for an `onyx` executable in `$ONYX_PATH/bin`, but with the Windows distribution, I forgot that is not how Onyx is shipped. This will need to be addressed in the next release.

For now, what happens if you move the `onyx_runtime.dll` file into the `bin` folder you created? Onyx needs that file to run and I believe it has to be in the same folder as `onyx.exe`. Does this fix the problem?

**Comment by magnetenstad** [comment]
Thank you for the quick response!

I can't seem to find any `.dll` files in `onyx-windows-wasmer-amd64.zip`. I guess this might be the reason onyx won't run either. (it builds to wasm just fine)

```
$ onyx run main.onyx
ERROR RESOLVING 'onyx_library_onyx_runtime'
Attempted to invoke imported function with no defintion, 'onyx_runtime.__file_get_standard'
```

The linux installation has the step "Please choose a WebAssembly runtime to use with your Onyx installation.". Am I missing the Windows counterpart?

**Comment by brendanfh** [comment]
No you did not miss anything. I should have been more diligent with checking the latest release before I published it. There was a small compilation error in the Windows build that didn't cause GitHub actions to say the build failed. This caused the `onyx_runtime.dll` file to not be compiled, and thus not be included in the zip file.

I have re-uploaded the corrected zip file made from #36 that *should* work for you now. You will need to nuked your current `C:/onyx` folder and replace it with the new version.

If that doesn't work I will reopen this issue. Have fun trying Onyx!

**Comment by magnetenstad** [comment]
After getting the dll, Onyx runs nicely on Windows.
However, the extension is unfortunately reporting the same errors as previously.
